packages/communication-react/review/stable/communication-react.api.md

```diff
@@ -1291,8 +1312,8 @@ export const createAzureCommunicationChatAdapter: ({ endpoint: endpointUrl, user
 // @public
 export const createAzureCommunicationChatAdapterFromClient: (chatClient: StatefulChatClient, chatThreadClient: ChatThreadClient) => Promise<ChatAdapter>;
 
-// @public
-export const createDefaultCallingHandlers: (callClient: StatefulCallClient, callAgent: CallAgent | undefined, deviceManager: StatefulDeviceManager | undefined, call: Call | undefined) => CallingHandlers;
+// @public (undocumented)
+export const createDefaultCallingHandlers: <AgentType extends CallAgent = CallAgent>(callClient: StatefulCallClient, callAgent: AgentType | undefined, deviceManager: StatefulDeviceManager | undefined, call: CallTypeOf<AgentType> | undefined) => CallHandlersOf<AgentType>;
```

Comment:
There are 2 ways of implement this function -
1. use Generic one but the default CallAgent will be just ACS CallAgent, the benefit of this solution is minimize the api we expose, and also reduce if-else code branch we need, but reading this api needs more time
2. use 2 functions, one for createDefaultCallingHandlers - which returns handlers only for ACS call, then createDefaulTeamstCallingHandlers - which only return handlers for Teams call, this will add up some complexities to number of functions, but api will look better than 1st one

Both of solution will be non-breaking change.

Reply:
I prefer (2) as it is easier for Contoso to understand and aligned with the factory functions you have planned for adapters. The return type can still be generic so that Contoso can write shared code on top.